### PR TITLE
Changes to allow storing of x-ray centring results

### PR DESCRIPTION
**Pull request by KarlLevik**
This also includes associated tests and minimal changes to the schema SQL files.

### CODE REVIEW

File: ispyb/sp/mxacquisition.py [hunk 1]
```diff
@@ -68,6 +68,17 @@ def __init__(self):
...
     StrictOrderedDict([('id',None), ('start_energy',None), ('end_energy',None), ('element',None), ('edge',None), ('r',None), ('g',None), ('b',None)])
 
+  _xray_centring_params =\
+   StrictOrderedDict([('id',None), ('grid_info_id',None), ('method',None), ('status',None), ('x',None), ('y',None)])
+
+  @classmethod
+  def get_xray_centring_result_params(cls):
+      return copy.deepcopy(cls._xray_centring_params)
+
+  def upsert_xray_centring_result(self, values):
```
Comment (Anthchirp) [inline ispyb/sp/mxacquisition.py]: Are you happy for me to change this to a keyword-argument format similar to [```upsert_program_ex```](https://github.com/DiamondLightSource/ispyb-api/blob/e8114b303753da7e0aecb53fec1cedc916a8fe64/ispyb/sp/mxprocessing.py#L126-L129)?

Comment (KarlLevik) [reply ispyb/sp/mxacquisition.py]: Absolutely - that sounds good.

File: conf/tables.sql [hunk 2]
```diff
@@ -281,7 +281,7 @@ DROP TABLE IF EXISTS `AutoProcProgramMessage`;
 /*!40101 SET @saved_cs_client     = @@character_set_client */;
 /*!40101 SET character_set_client = utf8 */;
 CREATE TABLE `AutoProcProgramMessage` (
-  `autoProcProgramMessageId` int(10) unsigned NOT NULL,
+  `autoProcProgramMessageId` int(10) unsigned NOT NULL AUTO_INCREMENT,
```
[the remark this answers is not in the document]
Comment (KarlLevik) [reply conf/tables.sql:284]: Yes - see https://github.com/DiamondLightSource/ispyb-database/blob/master/schema/updates/2018_11_14_AutoProcProgramMessage_autoinc.sql

**Comment by Anthchirp** [comment]
I've changed the function and added a fix for the new version of pytest that is running on Travis / Python 3.7.

**Pull request merged**